website/addons/base/__init__.py

```diff
@@ -864,6 +861,66 @@ def after_delete(self, node, user):
         """
         pass
 
+############
+# Archiver #
+############
+class GenericRootNode(object):
+    path = '/'
+    name = ''
+
+class StorageAddonBase(object):
```

Comment:
You may want to consider making this abstract and adding properties like `file_name` or `display_name`.
I'm not sure how that will interact with modular-odm however.